components/tidb_query/src/rpn_expr/impl_json.rs

```diff
@@ -14,6 +16,84 @@ fn json_type(arg: &Option<Json>) -> Result<Option<Bytes>> {
         .map(|json_arg| Bytes::from(json_arg.json_type())))
 }
 
+#[rpn_fn(varg)]
+#[inline]
+fn json_array(args: &[&Option<Json>]) -> Result<Option<Json>> {
+    Ok(Some(Json::Array(
+        args.iter()
+            .map(|json| match json {
+                None => Json::None,
+                Some(json) => json.to_owned(),
+            })
+            .collect(),
+    )))
+}
+
+fn json_validator(expr: &tipb::Expr) -> Result<()> {
```

Comment:
json_validator is not a good name. In fact this function is only useful for JSON_OBJECT.